sensor/common/compliance/nodescan_handler_impl.go

```diff
@@ -43,40 +59,44 @@ func (c *nodeScanHandlerImpl) ProcessMessage(_ *central.MsgToSensor) error {
 	return nil
 }
 
-func (c *nodeScanHandlerImpl) run() {
-	defer c.stoppedC.SignalWithError(c.stopC.Err())
-	defer close(c.toCentral)
-
-	for !c.stopC.IsDone() {
-		select {
-		case <-c.stopC.Done():
-			return
-		case scan, ok := <-c.nodeScans:
-			if !ok {
-				c.stopC.SignalWithError(errors.New("channel receiving node scans v2 is closed"))
+// run handles the messages from Compliance and forwards them to Central
+// This is the only goroutine that writes into the toCentral channel, thus it is responsible for creating and closing that chan
+func (c *nodeScanHandlerImpl) run() <-chan *central.MsgFromSensor {
+	toC := make(chan *central.MsgFromSensor)
+	go func() {
+		defer c.runFinishedC.Signal()
```

Comment:
Did you mean
```suggestion
		defer c.runFinishedC.SignalWithError(c.stopC.Err())
```
?

Reply:
I added this, but we have no guarantee that the  `c.stopC.Err()` will be something other than nil at the moment of evaluation of this statement.

Reply:
:-)
So this should be something like the following?

```go
defer func() {
    c.runFinishedC.SignalWithError(c.stopC.Err())
}()
```

Reply:
Let me add a test-case for this